addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobProvider.java

```diff
@@ -179,4 +189,45 @@ protected String getContentDispositionHeader(Blob blob) {
         return RFC2231.encodeContentDisposition(blob.getFilename(), false, null);
     }
 
+    @Override
+    public BlobStatus getStatus(ManagedBlob blob) throws IOException {
+        String key = stripBlobKeyPrefix(blob.getKey());
+        String objectKey;
+        String versionId;
+        int seppos = key.indexOf(VER_SEP);
+        if (seppos < 0) {
+            objectKey = key;
+            versionId = null;
+        } else {
```

Comment:
wdyt if we replace it by:
```
        String objectKey = key;
        String versionId = null;
        int seppos = key.indexOf(VER_SEP);
        if(seppos >= 0) {
            objectKey = key.substring(0, seppos);
            versionId = key.substring(seppos + 1);
        }
```

Reply:
I prefer the "static single assignment" form, where variables are assigned only once if possible and if it doesn't make the code unreadable.
Note that this pattern helps lambdas as well (not in this case, there's no lambda) as it makes the variables effectively-final.